src/idl_parser.cpp

```diff
@@ -267,7 +267,8 @@ T *LookupTableByName(const SymbolTable<T> &table, const std::string &name,
   TD(StringConstant, 257, "string constant") \
   TD(IntegerConstant, 258, "integer constant") \
   TD(FloatConstant, 259, "float constant") \
-  TD(Identifier, 260, "identifier")
+  TD(NumericConstant, 260, "nan, inf or function name (signed)") \
```

Comment:
This will cause someone writing a schema field like `inf:string` to get a pretty confusing error? If they intended to use `inf` as short for `information` or whatever :)

Might it be better to keep it as `Identifier` and explicitly recognize the few identifiers we care about only when parsing values (not while parsing field names)?

Reply:
> This will cause someone writing a schema field like inf:string to get a pretty confusing error?

No. In this case `inf` will be the identifier. If someone writes `+inf` the parser will return an error that it expects `Identifier` token. This case added into tests.

I tried to find another solution but couldn't.
Method `ParseAnyValue` is recursive so it is impossible to detect which part is now parsed, identifier or value.
I tried to solve it with flags on the assignment expressions (`=` for scheme and `:` for JSON) but it looks terrible. 
Maybe you have an idea where these keywords can be matched as values?

Reply:
I am not sure I follow the problem. It would seem `+inf` currently gets parsed as a `kTokenStringConstant` ?

>         const auto has_sign = (c == '+') || (c == '-');
>         // '-'/'+' and following identifier - can be a predefined constant like:
>         // NAN, INF, PI, etc or it can be a function name like cos/sin/deg.
>         if (IsIdentifierStart(c) || (has_sign && IsIdentifierStart(*cursor_))) {
>           // Collect all chars of an identifier:
>           const char *start = cursor_ - 1;
>           while (IsIdentifierStart(*cursor_) || is_digit(*cursor_)) cursor_++;
>           attribute_.append(start, cursor_);
>           token_ = has_sign ? kTokenStringConstant : kTokenIdentifier;
>           return NoError();
>         }

No idea why this was done. What would happen if we recognized these during `Next` and made them into `kTokenFloatContant`?

Reply:
I removed the introduced kTokenNumericConstant.
Now all processing is located inside the `Parser::ParseSingleValue()`. It looks much better and it passed all tests.
Could you review this new solution?